libs/blocks/preflight/panels/seo.js

```diff
@@ -0,0 +1,223 @@
+const bodyResult = signal({ icon: DEF_ICON, title: 'Body size', description: DEF_DESC });
+const loremResult = signal({ icon: DEF_ICON, title: 'Lorem Ipsum', description: DEF_DESC });
+const linksResult = signal({ icon: DEF_ICON, title: 'Links', description: DEF_DESC });
+let Passed = true;
+
+function checkH1s() {
+  const h1s = document.querySelectorAll('h1');
+  const result = { ...h1Result.value };
+  if (h1s.length === 1) {
+    result.icon = 'green';
+    result.description = 'Only one H1 on the page.';
+  } else {
+    Passed = false;
+    result.icon = 'red';
+    if (h1s.length > 1) {
+      result.description = 'Reason: More than one H1 on the page.';
+    } else {
+      result.description = 'Reason: No H1 on the page.';
+    }
+  }
+  h1Result.value = result;
+}
+
+async function checkTitle() {
+  const titleSize = document.title.replace(/\s/g, '').length;
+  const result = { ...titleResult.value };
+  if (titleSize < 15) {
+    Passed = false;
+    result.icon = 'red';
+    result.description = 'Reason: Title size is too short.';
+  } else if (titleSize > 70) {
+    Passed = false;
+    result.icon = 'red';
+    result.description = 'Reason: Title size is too long.';
+  } else {
+    result.icon = 'green';
+    result.description = 'Title size is good.';
+  }
+  titleResult.value = result;
+}
+
+async function checkCanon() {
+  const canon = document.querySelector("link[rel='canonical']");
+  const result = { ...canonResult.value };
+  const { href } = canon;
+  const resp = await fetch(href, { method: 'HEAD' });
+  if (!resp.ok) {
+    Passed = false;
+    result.icon = 'red';
+    result.description = 'Reason: Error with canonical reference.';
+  }
+  if (resp.ok) {
+    if (resp.status >= 300 && resp.status <= 308) {
+      Passed = false;
+      result.icon = 'red';
+      result.description = 'Reason: Canonical reference redirects.';
+    } else {
+      result.icon = 'green';
+      result.description = 'Canonical referenced is valid.';
+    }
+  }
+  canonResult.value = result;
+}
+
+async function checkDescription() {
+  const metaDesc = document.querySelector('meta[name="description"]');
+  const result = { ...descResult.value };
+  if (!metaDesc) {
+    Passed = false;
+    result.icon = 'red';
+    result.description = 'Reason: No meta description found.';
+  } else {
+    const descSize = metaDesc.content.replace(/\s/g, '').length;
+    if (descSize < 50) {
+      Passed = false;
+      result.icon = 'red';
+      result.description = 'Reason: Meta description too short.';
+    } else if (descSize > 150) {
+      result.icon = 'red';
+      result.description = 'Reason: Meta description too long.';
+    } else {
+      result.icon = 'green';
+      result.description = 'Meta description is good.';
+    }
+  }
+  descResult.value = result;
+}
+
+async function checkBody() {
+  const result = { ...bodyResult.value };
+  const { length } = document.documentElement.innerText;
+
+  if (length > 100) {
```

Comment:
It might be nice to save some of these conditionals, like the `length` limit here or the `descSize` limits on lines 83 and 150  as `CONST` vars or part of a config to increase readability and change easily if need be.